samples/cpp/calibration.cpp

```diff
@@ -529,6 +543,40 @@ int main( int argc, char** argv )
             case ASYMMETRIC_CIRCLES_GRID:
                 found = findCirclesGrid( view, boardSize, pointbuf, CALIB_CB_ASYMMETRIC_GRID );
                 break;
+            case CHARUCOBOARD:
+            {
+                cv::aruco::Dictionary dictionary;
+                if (dictFilename == "None") {
+                    dictionary = cv::aruco::getPredefinedDictionary(arucoDict);
+                }
+                else {
+                    cv::FileStorage dict_file(dictFilename, cv::FileStorage::Mode::FORMAT_JSON && cv::FileStorage::Mode::READ);
+                    cv::FileNode fn(dict_file.root());
+                    dictionary.readDictionary(fn);
+                }
+                cv::Ptr <cv::aruco::CharucoBoard> board =
+                    cv::makePtr<cv::aruco::CharucoBoard>(cv::aruco::CharucoBoard({ boardSize.width + 1,boardSize.height + 1 },
```

Comment:
No need to use `cv::Ptr` for board. You can allocate it on stack as local variable.